### PR TITLE
helm: Add updater support to `teleport-kube-agent` chart

### PR DESCRIPTION
**Pull request by hugoShaka**
This PR adds updater support to the `teleport-kube-agent` chart.

Cloud users will need to set

```yaml
updater:
  enabled: true
```

While on-prem users will also need to set their own version server

```yaml
updater:
  enabled: true
  versionServer: https://foo.example.com
```

By default the updater deployment will honour all applicable Kubernetes-related values like `resources`, `affinity`, `extraEnv`, ... but those can be overwritten like we do in `teleport-cluster`:

```yaml
updater:
  enabled: true
  resources:
    requests:
      cpu: "0.5"
      memory: 1Gi
```

### CODE REVIEW

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx [hunk 1]
```diff
@@ -106,6 +106,90 @@ This parameter is not mandatory to preserve backwards compatibility with older c
...
+    limits: ~
+```
+
+Other updater-specific values that can be defined in `updater` are described
+below:
+
+### `updater.enabled`
+
+Enables the Kube Agent Updater and deploys it alongside the Teleport Agent.
+You must enable this when:
```
Comment (r0mant) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:143]: Nit: I think "must" is a bit too strong a word here and below. I would rephrase to "updater should be enabled ...".

File: examples/chart/teleport-kube-agent/values.yaml [hunk 2]
```diff
@@ -136,6 +136,18 @@ tls:
   # The filename inside the secret is important - it _must_ be ca.pem
   existingCASecretName: ""
 
+updater:
+  enabled: false
+  # `updater.versionServer` is the URL of the version server the agent fetches
+  # the target version from.
+  versionServer: "https://update.gravitational.io/v1/"
+  # Release channel the agent subscribes to.
+  releaseChannel: "stable/cloud"
```
Comment (r0mant) [inline examples/chart/teleport-kube-agent/values.yaml:145]: TBH I'm not sure why we split versionServer and releaseChannel in 2 parameters. Self-hosted users would need to change both anyway and it's not very clear that "channel" is actually just a path on the version server. Wouldn't having just a single parameter make things simpler and clearer?

```yaml
versionServer: https://update.gravitational.io/v1/stable/cloud
```

Comment (hugoShaka) [reply examples/chart/teleport-kube-agent/values.yaml:145]: > TBH I'm not sure why we split versionServer and releaseChannel in 2 parameters. Self-hosted users would need to change both anyway and it's not very clear that "channel" is actually just a path on the version server. Wouldn't having just a single parameter make things simpler and clearer?

I suspect having a dedicated release channel will be very handy when we will want to migrate existing cloud users into automatic upgrades. We will face strong coupling issues:
- a tenant can be enrolled into automatic upgrades if the majority of their agents are automatically upgraded
- we can't bump the target version of the cloud/stable channel if it will break at least a tenant

To support a seamless transition I think we will have to create per-tenant release channels. In this scenario, it will be less awkward for users to only specify their tenant-specific release channel while still using the default versionServer.

Comment (r0mant) [reply examples/chart/teleport-kube-agent/values.yaml:145]: I don't think we'll want to maintain per-tenant versions (sounds like a huge burden) but I understand that this allows you to change either server URL or version file path independently.

I think it's fine, my main issue was that "release channel" is just quite confusing term for what effectively is just a path component of the URL. Can we at least update the comment explaining that the updater will basically just call `filepath.Join(versionServer, releaseChannel)` to fetch the version information?

**Review by r0mant — APPROVED** [review]

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx [hunk 3]
```diff
@@ -106,6 +106,90 @@ This parameter is not mandatory to preserve backwards compatibility with older c
...
+- using Teleport Cloud and you tenant is enrolled into automatic updates.
```
Comment (marcoandredinis) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:144]:
```suggestion
- using Teleport Cloud and your tenant is enrolled into automatic updates.
```

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx [hunk 4]
```diff
@@ -106,6 +106,90 @@ This parameter is not mandatory to preserve backwards compatibility with older c
...
+  (You can check this through the web UI, choose "add a new node" of type
```
Comment (marcoandredinis) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:145]: :nail_care: 
`Add Kubernetes` / `Enroll New Resource of type Kubernetes`

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx [hunk 5]
```diff
@@ -106,6 +106,90 @@ This parameter is not mandatory to preserve backwards compatibility with older c
...
+  Kubernetes, and check if the value is turned on.)
+- using Teleport on-prem and you maintain your own version server.
```
Comment (marcoandredinis) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:147]: Can we provide an example of such server?
Even if it only returns static values
A simple `net/http` based program should be enough

File: examples/chart/teleport-kube-agent/.lint/updater.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+proxyAddr: proxy.example.com:3080
+roles: ""
+updater:
+  enabled: true
+  versionServer: https://my-custom-version-server/v1
+  releaseChannel: cloud/preview
```
Comment (marcoandredinis) [inline examples/chart/teleport-kube-agent/.lint/updater.yaml:6]: Can we have the default value here? or something which says `custom`?
Otherwise, may lead to people assuming there's an official `cloud/preview` channel (even if the versionServer is clearly _custom_)

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx (hunk 5 above)
Comment (hugoShaka) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:147]: 👍 , I have an issue to write documentation on how to host your own version server. Will address this in an other PR.

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx (hunk 1 above)
Comment (hugoShaka) [reply docs/pages/reference/helm-reference/teleport-kube-agent.mdx:143]: I think the docs folks don't like should (at least I got the feedback multiple times). I can change this from a MUST to a CAN.

**Review by marcoandredinis — APPROVED** [review]
